assembl/static2/js/app/components/debate/thread/nuggets.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import range from 'lodash/range';
+
+import { getDomElementOffset, setDomElementOffset } from '../../../utils/globalFunctions';
+
+class Nuggets extends React.Component {
+  static get SPACER_SIZE() {
+    // Vertical space between overflowing extracts in px
+    return 100;
+  }
+
+  static completeLevelFromId(id) {
+    return id.split('-').slice(1).map((ident) => {
+      return Number(ident);
+    });
+  }
+
+  static completeLevel(rowIndex, fullLevel) {
+    return fullLevel ? [rowIndex, ...fullLevel] : [rowIndex];
+  }
+
+  static completeLevelGreater(a, b) {
+    return range(Math.min(a.length, b.length)).reduce((result, index) => {
+      if (result !== null) return result;
+      if (a[index] === b[index] && (a.length === index + 1 || b.length === index + 1)) return a.length > b.length;
+      return a[index] !== b[index] ? a[index] > b[index] : result;
+    }, null);
+  }
+
+  static previousExtract(completeLevel, otherExtracts) {
+    return !otherExtracts || !otherExtracts.length
+      ? null
+      : otherExtracts.reduce((prevExtract, extract) => {
+        const currentCompleteLevel = prevExtract ? Nuggets.completeLevelFromId(prevExtract.getAttribute('id')) : null;
```

Comment:
So you're using the DOM id to fetch the previous nugget. But you have a react Ref. Why not use the `this.node` to find the component (Maybe you will have to keep a state with the name of the component so you can easily find the node). Check to see if this is feasable. Perhaps the methods `completeLevelFromId` and `completeLevel` might also not be needed with this method.

Reply:
I think I can do this but walking through the dom doesn't seems very resilient (if a component structure changes)